events/nats.go

```diff
@@ -225,32 +222,20 @@ func (n *NatsJetstream) addConsumer() error {
 	return nil
 }
 
-// PublishAsyncWithContext publishes an event onto the NATS Jetstream.
-func (n *NatsJetstream) PublishAsyncWithContext(_ context.Context, resType ResourceType, eventType EventType, objID string, obj interface{}) error {
+// Publish publishes an event onto the NATS Jetstream. The caller is responsible for message
+// addressing and data serialization.
+func (n *NatsJetstream) Publish(_ context.Context, subject string, data []byte) error {
```

Comment:
I think its worth still having a defined `Payload` struct with minimum fields in this package that the callers pass into this method. This will ensure all controllers follow the same format without it being left for interpretation. 
If really required we can have another method just does `PublishBytes` 

```go

type Payload struct {
    // ResourceType is the kind of the object affected or included in Data.
    ResourceType ResourceType // server | condition
    
    // EventType is the kind of event this is.
    EventType EventType // create | update | delete

    // Data to be published
    Data []byte  // server | condition
}

```

Reply:
Based on the offline discussion it makes sense that the message struct does not live in this library.